### PR TITLE
Don't finalize if status or list command

### PR DESCRIPTION
**Pull request by mike-sul**
don't finalize a pending installation if `list` or `status` command is called.

### CODE REVIEW

**Comment by doanac** [comment]
We should pull in @mike-scott. I think he has customers that sometimes do the following sequence:

 * aktualizr-lite update 
 * the ostree hash has changed, so reboot
 * aktualizr-lite status # "completes" the install

**Comment by mike-sul** [comment]
> We should pull in @mike-scott. I think he has customers that sometimes do the following sequence:
> 
> * aktualizr-lite update
> * the ostree hash has changed, so reboot
> * aktualizr-lite status # "completes" the install

yep, I know that it became a feature bug. I suppose we can introduce another command as you did recently in the prototype for customer X, i.e. `aktualizr-lite finalize`?

**Comment by doanac** [comment]
i think an explict `fiinalize` subcommand makes sense. maybe add that in here, and then @mike-scott  can just inform people that need to know?

**Comment by mike-sul** [comment]
@doanac @mike-scott Added `aktualizr-lite finalize` command.

**Review by doanac — COMMENTED** [review]
just have one question. looks good though.

File: src/compose/composeappmanager.cc [hunk 1]
```diff
@@ -287,8 +276,20 @@ data::InstallationResult ComposeAppManager::install(const Uptane::Target& target
...
+    auto need_start_flag = cfg_.apps_root / app_name / Docker::ComposeAppEngine::NeedStartFile;
+    if (boost::filesystem::exists(need_start_flag)) {
+      if (ir.result_code == data::ResultCode::Numeric::kOk) {
+        app_engine_->run({app_pair.first, app_pair.second});
+      }
+      boost::filesystem::remove(need_start_flag);
+    }
+  }
+
+  if (ir.result_code == data::ResultCode::Numeric::kOk) {
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply src/compose/composeappmanager.cc:292]: > Should we report the container details unconditionally?

The report event is not sent unconditionally. So, what's the point to inquire dockerd and fill a context of the event which is never sent? (see finalizeInstallation method).
Although the following condition is slightly better, so we send info on running apps in both cases, OK, and if a rollback happens.

`
if (data::ResultCode::Numeric::kNeedCompletion != update_finalization_result.result_code.num_code) {
`

[the remark this answers is not in the document]
Comment (mike-sul) [reply src/compose/composeappmanager.cc]: Updated via the forced push.
BTW, we've forgotten to refactor/adjust `ComposeAppManager::containerDetails()` so it uses DockerClient that @ldts introduced.

**Review by doanac — APPROVED** [review]

File: src/liteclient.cc [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include "composeappmanager.h"
 #include "crypto/keymanager.h"
 
-LiteClient::LiteClient(Config& config_in, const AppEngine::Ptr& app_engine)
+LiteClient::LiteClient(Config& config_in, const AppEngine::Ptr& app_engine, bool finalize)
```
Comment (ldts) [inline src/liteclient.cc:13]: what about adding the flag to the Config as an action?

**Pull request merged**